src/apps/frontend/types/async-operation.ts

```diff
@@ -1,27 +1,28 @@
-import { JsonObject } from './common-types';
+import { JsonObject, Nullable } from './common-types';
 
 export interface AsyncError {
   code: string;
   message: string;
 }
 
 export interface AsyncResult<T> {
-  error?: AsyncError;
-  data?: T;
+  error?: Nullable<AsyncError>;
+  data?: Nullable<T>;
 }
 
-export interface UseAsyncResponse<T> extends AsyncResult<T> {
-  result: T;
-  asyncCallback: (...args: unknown[]) => Promise<T>;
+export interface UseAsyncResponse<T> {
+  result: Nullable<T>;
+  asyncCallback: (...args: unknown[]) => Promise<Nullable<T>>;
   isLoading: boolean;
+  error: Nullable<AsyncError>;
 }
 
 export class AsyncOperationError implements AsyncError {
   code: string;
   message: string;
 
   constructor(json: JsonObject) {
-    this.code = json.code as string;
-    this.message = json.message as string;
+    this.code = (json.code as string) ?? 'UNKNOWN_ERROR';
```

Comment:
Don't think this is right.

We are changing the behaviour of the code i.e. defaulting to something to address typechecks.

Are code and message required property? If yes this should not happen. If not, then what is the behaviour on the upstream when they are not defined (from the typedef in `AsyncOperationError`, they do not seem to be undefined)